Server/Ebenezer/User.cpp

```diff
@@ -8690,6 +8690,16 @@ int CUser::GetNumberOfEmptySlots() const
 	return emptySlotCount;
 }
 
+bool CUser::CheckExistEvent(int questId, int questState) const
+{
+	for (const auto& quest : m_pUserData->m_quests)
+	{
+		if (quest.sQuestID == questId && quest.byQuestState == questState)
```

Comment:
The quest state check here should decide the result; if it encounters the quest ID and it doesn't match, it shouldn't keep processing. It should just return immediately either way.